eslint.config.mjs

```diff
@@ -0,0 +1,467 @@
+    languageOptions: {
+      globals: {
+        ...globals.node,
+        ...globals.jasmine,
+      },
+    },
+    rules: {
+      ...jasmine.configs.recommended.rules,
+      "jasmine/new-line-before-expect": "off",
+      "jasmine/new-line-between-declarations": "off",
+      "jasmine/no-focused-tests": "error",
+      "jasmine/no-pending-tests": "off",
+      "jasmine/no-spec-dupes": ["error", "branch"],
+      "jasmine/no-suite-dupes": ["error", "branch"],
+      "jasmine/prefer-jasmine-matcher": "off",
+      "jasmine/prefer-toHaveBeenCalledWith": "off",
+    },
+  },
+  {
+    files: jsFiles("test/fuzz"),
+    rules: {
+      "import/no-unresolved": [
+        "error",
+        { ignore: [".*/build/image_decoders/.*"] },
+      ],
+    },
+  },
+  {
+    files: jsFiles("test/unit"),
+    rules: {
+      "import/no-unresolved": ["error", { ignore: ["pdfjs/"] }],
+    },
+  },
+  {
+    files: jsFiles("test/integration"),
+    rules: {
+      "no-restricted-syntax": [
+        "error",
+        {
+          selector: "CallExpression[callee.name='waitForTimeout']",
+          message:
+            "`waitForTimeout` can cause intermittent failures and should not be used (see issue #17656 for replacements).",
+        },
+      ],
+    },
+  },
+
+  /* ======================================================================== *\
+                            External libraries
+  \* ======================================================================== */
+
+  {
+    files: jsFiles("external"),
+
+    languageOptions: { globals: globals.node },
+  },
+
+  /* ======================================================================== *\
+                             Examples
+  \* ======================================================================== */
+
+  {
+    files: jsFiles("examples"),
+
+    languageOptions: {
+      globals: {
+        pdfjsImageDecoders: false,
+        pdfjsLib: false,
+        pdfjsViewer: false,
+      },
+    },
+  },
+  {
+    files: [...jsFiles("examples/node"), ...jsFiles("examples/webpack")],
+
+    languageOptions: { globals: globals.node },
+  },
+
+  /* ======================================================================== *\
+                            Chromium extension
+  \* ======================================================================== */
+
+  {
+    files: jsFiles("extensions/chromium"),
+
+    languageOptions: {
+      globals: globals.webextensions,
+      sourceType: "script",
+    },
+
+    rules: {
+      "no-var": "off",
+    },
+  },
+  {
+    files: chromiumExtensionServiceWorkerFiles,
+
+    languageOptions: {
+      globals: globals.serviceworker,
+      sourceType: "script",
+    },
+  },
+
+  /* ======================================================================== *\
+                                Other
+  \* ======================================================================== */
+  {
+    files: ["gulpfile.mjs"],
+    languageOptions: { globals: globals.node },
+  },
```

Comment:
Doesn't this mean that we could/should remove the following line?
https://github.com/mozilla/pdf.js/blob/7a962031e90d3b19604f430dae9d4df4712c12f7/gulpfile.mjs#L15